app/models/image.rb

```diff
@@ -12,6 +12,18 @@ class Image < BaseModel
   FAILED = 'failed'.freeze
   COMPLETE = 'complete'.freeze
 
+  PROFILE = 'profile'.freeze
+  THUMBNAIL = 'thumbnail'.freeze
+  PURPOSES = [PROFILE, THUMBNAIL].freeze
+
+  def self.profile
```

Comment:
So you're adding the `purpose` field to every image type, then?

Also, is profile-vs-thumbnail a generic enough relation to warrant having them in every image sub-type?  (Mostly playing devil's advocate here)